web/frontend/src/components/slam/index.svelte

```diff
@@ -118,11 +124,47 @@
     }
   };
 
+  // TODO: Why is this running twice per iteration?
+  const refreshPaths = async () => {
+    try {
+      refreshErrorMessagePaths = undefined;
+      const res = await motionClient.getPlan({
+        namespace: "rdk",
+        type: "component",
+        subtype: "base",
+        name: bases[0]!.name,
+      }, true)
+      if (res.currentPlanWithStatus?.status?.state === motionApi.PlanState.PLAN_STATE_IN_PROGRESS) {
+        executionID = res.currentPlanWithStatus.plan.executionId;
+        motionPath = res.currentPlanWithStatus.plan.stepsList.map(({stepMap: [[, map]]}) => {
+          // TODO: I don't know how to convince the TS typesystem that map is always has a pose.
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
```

Comment:
@njooma @micheal-parks & @ethanlook I'm not sure how to placate the typescript linter without disabling it as I have done here.

Reply:
I would use optional chaining and the nullish coallescing operator to resolve this. Is it that the map is sometimes undefined, or the pose?

If the map is sometimes undefined, then

```
return `${map?.pose.x ?? 0},${map?.pose.y ?? 0}`;
```

If the pose is sometimes undefined, then

```
return `${map.pose?.x ?? 0},${map.pose?.y ?? 0}`;
```

Or a combination of both, if they are both possibly undefined.

Reply:
Thanks @ethanlook ! I don't really understand why all these types can possibly be undefined based on the proto definition. I've updated the code with a version that passes typechecking without needing to disable linting. Can you please let me know if the current solution is acceptable or if there is a more ideomatic version out there?

We don't want to return any string if any of the nodes of the tree are undefined.

Reply:
I'm hoping that we can come up with something a bit better since we're likely performing these checks unnecessarily as you suggest, I'll have some time today to pull this down and play with it, will give a suggestion once I'm done

Reply:
Thank you very much @micheal-parks!

Reply:
Although I'm not happy with how many optionals still exist here (and I'm not exactly sure why these protos are being marked as optional, that's a problem for someone who understands our protobuf tooling better), this is how I would rewrite this to make it a bit simpler and more readable:

```ts
const paths: string[] = []

for (const { stepMap: [stepMap] } of res.currentPlanWithStatus.plan?.stepsList ?? []) {
  const { pose } = stepMap?.[1] ?? {}

  if (pose) {
    paths.push(`${pose.x},${pose.y}`);
  }
}

motionPath = paths.join('\n');
```

We're currently and unfortunately using a pretty awful tool for proto to JS generation, so this may just be something we have to live with. We've mitigated the bad types and api response shapes in the past by just doing data transforms in the typescript SDK.

Reply:
There's another problem here, and it's that we're building this feature on top of a backdoor that I added for internal testing. I'd argue here that the data structure is wrong, and that we should be passing the component a `Float32Array` instead of a string.

The only reason it accepts a string was because the backdoor debug feature parsed `.txt` files. Binary would be much better for this data for many reasons. 

Not sure if we want to change this now, but I'd like to acknowledge that we're building this on top of something that was never intended for this.

Reply:
Thanks @micheal-parks ! I'll try out your suggestion & get started on trying to refactor this to not use the back door.

Reply:
@nicksanford My recommended route would be to change `<SlamMap2D>` so that `motionPath` is of type `Float32Array | undefined`, and then update this component so that both the API response and the `.txt` drag and drop debug feature transform their data into a `new Float32Array([x1, y1, x2, y2, x3, y3, ...])`.

This will put the `<SlamMap2D>` component in the most robust state for the future IMO. In my experience using objects or structs for a (potentially large) series of points / vectors / etc. isn't sustainable and eventually replaced with continuous typed array data.

Reply:
@micheal-parks 
I've added your suggestions in terms of handling the nullability more idomatically.

Given there is some urgency from the slam team on being able to visualize the paths ASAP, would it be acceptable if I were to do that type refactor as an immediate follow-up to this PR? I'll make sure it gets done before we move on from this project.

Reply:
@micheal-parks Tickets I've created to track those follow ups:
https://viam.atlassian.net/browse/RSDK-6434
https://viam.atlassian.net/browse/RSDK-6435

Reply:
That works for me! Thank you for adding those follow ups 🙏